src/bot.ts

```diff
@@ -106,14 +99,61 @@ Also, here are some links you might find useful:
   )
 }
 
-function _handleFetchDataError (error: Error): string {
-  log.error(error)
-  return 'N/A'
-}
-
-dfusionRepo.watchOrderPlacement({
-  onNewOrder (event) {
-    bot.sendMessage(channelId, formatNewOrders(event))
+dfusionService.watchOrderPlacement({
+  onNewOrder (order) {
+    const {
+      // owner,
+      buyToken,
+      sellToken,
+      validFrom,
+      validUntil,
+      // validFromBatchId,
+      // validUntilBatchId,
+      priceNumerator,
+      priceDenominator
+      // event
+    } = order
+
+    // Calculate the price
+    let price
+    if (buyToken.decimals >= sellToken.decimals) {
+      const precisionFactor = 10 ** (buyToken.decimals - sellToken.decimals)
+      price = priceNumerator.dividedBy(priceDenominator.multipliedBy(precisionFactor))
+    } else {
+      const precisionFactor = 10 ** (sellToken.decimals - buyToken.decimals)
+      price = priceNumerator.multipliedBy(precisionFactor).dividedBy(priceDenominator)
+    }
+
+    // Label for token
+    // TODO: to use the shared utils function when available safeTokenName
+    const sellTokenLabel = sellToken.symbol || sellToken.name || sellToken.address
+    const buyTokenLabel = buyToken.symbol || buyToken.name || buyToken.address
+
+    // Only display the valid from if the period hasn't started
+    const now = new Date()
+    let datesDescription = ''
+    if (validFrom > now) {
+      // The order is not active yet
+      datesDescription = `  - *Tradable*: \`${moment(validFrom).calendar()} GMT\`, \`${moment(validFrom).fromNow()}\`\n`
+    }
+    datesDescription += `  - *Expires*: \`${moment(validUntil).calendar()} GMT\`, \`${moment(validUntil).fromNow()}\``
+
+    // Format the amounts
+    const sellAmountFmt = sellToken
```

Comment:
can `sellToken` ever be `{}` here? if so i'd watch out as it's always `truthy` but otherwise it's fine. Btw are you on `strict` TS?

Reply:
Nup, initially was either null or a fine object.

> Btw are you on strict TS?

I have `strictNullChecks` if that's what u mean